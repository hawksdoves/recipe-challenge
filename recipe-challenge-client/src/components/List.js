import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from 'semantic-ui-react';

import Recipe from './Recipe';

function List(props) {
    const { recipes } = props

    return (
        <div>
            <Card.Group>
            {recipes.map(recipe => {
                return (
                    
                            <Link to={`/recipe/${recipe.id}`} key={`Link-${recipe.id}`} className='card'>
                                <Recipe recipe={recipe} key={recipe.id} />
                            </Link>
                 
                )
            })}
            </Card.Group>
        </div>
    )
    
};

export default List;