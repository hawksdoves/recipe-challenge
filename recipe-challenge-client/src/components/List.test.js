import React from 'react';
import { shallow } from 'enzyme';

import List from './List';

describe('List', () => {
    it('should render all the recipes correctly', () =>{
        const recipes = [
            {
                id: '1',
                ingredients: [{
                    name: 'steak',
                    quantity: '3'
                }]
            },
            {
                id: '2',
                ingredients: [{
                    name: 'chicken',
                    quantity: '2'
                }]
            }
        ]
        const wrapper = shallow(<List recipes={recipes} />);

        expect(wrapper).toMatchSnapshot()
    })
});
