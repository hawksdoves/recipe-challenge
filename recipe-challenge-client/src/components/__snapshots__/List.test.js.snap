// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render all the recipes correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <CardGroup>
        <Link
            className="card"
            replace={false}
            to="/recipe/1"
        >
            <Recipe
                recipe={
                    Object {
                        "id": "1",
                        "ingredients": Array [
                          Object {
                            "name": "steak",
                            "quantity": "3",
                          },
                        ],
                      }
                }
            />
        </Link>
        <Link
            className="card"
            replace={false}
            to="/recipe/2"
        >
            <Recipe
                recipe={
                    Object {
                        "id": "2",
                        "ingredients": Array [
                          Object {
                            "name": "chicken",
                            "quantity": "2",
                          },
                        ],
                      }
                }
            />
        </Link>
    </CardGroup>
</div>,
  "nodes": Array [
    <div>
      <CardGroup>
            <Link
                  className="card"
                  replace={false}
                  to="/recipe/1"
            >
                  <Recipe
                        recipe={
                              Object {
                                    "id": "1",
                                    "ingredients": Array [
                                      Object {
                                        "name": "steak",
                                        "quantity": "3",
                                      },
                                    ],
                                  }
                        }
                  />
            </Link>
            <Link
                  className="card"
                  replace={false}
                  to="/recipe/2"
            >
                  <Recipe
                        recipe={
                              Object {
                                    "id": "2",
                                    "ingredients": Array [
                                      Object {
                                        "name": "chicken",
                                        "quantity": "2",
                                      },
                                    ],
                                  }
                        }
                  />
            </Link>
      </CardGroup>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <List
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "recipes": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <CardGroup>
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/1"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "1",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "steak",
                                                                "quantity": "3",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/2"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "2",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "chicken",
                                                                "quantity": "2",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
          </CardGroup>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <CardGroup>
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/1"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "1",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "steak",
                                                                "quantity": "3",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
                    <Link
                              className="card"
                              replace={false}
                              to="/recipe/2"
                    >
                              <Recipe
                                        recipe={
                                                  Object {
                                                            "id": "2",
                                                            "ingredients": Array [
                                                              Object {
                                                                "name": "chicken",
                                                                "quantity": "2",
                                                              },
                                                            ],
                                                          }
                                        }
                              />
                    </Link>
          </CardGroup>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    recipes={
        Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
            },
          ]
    }
/>,
}
`;
